apps/dashboard/pkg/migration/conversion/conversion.go

```diff
@@ -9,76 +12,185 @@ import (
 	dashv1 "github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v1beta1"
 	dashv2alpha1 "github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v2alpha1"
 	dashv2beta1 "github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v2beta1"
+	"github.com/grafana/grafana/apps/dashboard/pkg/migration"
+	"github.com/grafana/grafana/apps/dashboard/pkg/migration/schemaversion"
 )
 
 var logger = logging.DefaultLogger.With("logger", "dashboard.conversion")
 
+// withConversionMetrics wraps a conversion function with metrics and logging for the overall conversion process
+func withConversionMetrics(sourceVersionAPI, targetVersionAPI string, conversionFunc func(a, b interface{}, scope conversion.Scope) error) func(a, b interface{}, scope conversion.Scope) error {
```

Comment:
What about extracting this function into a different file? So we keep this one as registry :)